test/util.js

```diff
@@ -183,7 +183,8 @@ async function prepareUnitTest(model, outputDirectory, parameters = {}) {
         await checkTranspilation(tsFiles)
     }
     configuration.setFrom(configurationBefore)
-    return { astw: new ASTWrapper(path.join(parameters.fileSelector(paths), 'index.ts')), paths }
+    // return undefined for astw in case type-generation failed
+    return { astw: paths?.length > 0 ? new ASTWrapper(path.join(parameters.fileSelector(paths), 'index.ts')) : undefined, paths }
```

Comment:
that was a "nice" (read: hard to love) way of finding when tests failed due to compilation errors, i.e. when the test model contained syntax errors, etc.
What are the implications of returning `undefined` here? Maybe we should take this opportunity to make the fail-state implicit and have something like

```suggestion
    if (!paths?.length) throw new Error('type-generation failed')
    return { astw: new ASTWrapper(path.join(parameters.fileSelector(paths), 'index.ts')), paths }
```
?

Reply:
Yes, in retrospect, throwing an error is clearer and would also work for the negative test case in `drafts.test.js`

```js
        await expect(
            prepareUnitTest('draft/error-catalog-service.cds', locations.testOutput('bookshop_projection'), {
                typerOptions: { logLevel: 'ERROR' },
            })
        ).rejects.toThrow('type-generation failed')
```

Reply:
We could also just remove the `.catch` clause for the `cds2ts` call. This would propagate the thrown errors inside csd-typer to the caller of `prepareUnitTest`

Reply:
Now that I think about it, I believe there was a deeper reason for just having an implicit fail 🤔
I seem to remember that explicitly throwing or removing `.catch` would just halt the entire test suite on the first failing test, instead of showing all failing tests. But I don't remember for sure...

Reply:
Hm. I made a few tests where I broke some cds models in `/test/unit/files`, but only those were failing then. The rest of the test suites ran without any issues

Reply:
Well, regardless, I think your suggested changes are objectively better than having arcane tribal knowledge of why tests should fail in certain constellations. So I guess as long as the tests run as expected, this is an improvement. 👍